modules/social_features/social_post/js/keycode-submit.js

```diff
@@ -10,29 +10,24 @@
   Drupal.behaviors.keycodeSubmit = {
     attach: function (context, settings) {
 
-      // Enable CTRL+Enter to submit a form.
-      var keySubmit = function (form, textarea, submit) {
-        textarea.on("keydown", function (e) {
-          // Make sure the textarea contains text.
-          if ($.trim(textarea.val()) != "") {
-            if ((e.keyCode == 13 && e.ctrlKey) || (e.keyCode == 13 && e.metaKey)){
-              e.preventDefault();
-              submit.prop('disabled', true);
-              form.submit();
-            }
-          }
-        });
-      }
-
       // Post form.
-      var postForm = $('#social-post-entity-form');
-      keySubmit($(postForm), $('.form-textarea', postForm), $('.form-submit', postForm));
-
-      // Comment forms.
-      $('.comment-form').each(function () {
-        keySubmit($(this), $('.form-textarea', this), $('.form-submit', this));
-      });
-
+        $('#social-post-entity-form .form-textarea')
+      .off('keydown', onTextAreaKeyDown)
+      .on('keydown', onTextAreaKeyDown);
+      // comment form
+        $('.comment-form .form-textarea').on('keydown')
+      .off('keydown', onTextAreaKeyDown)
+      .on('keydown', onTextAreaKeyDown);
     }
   };
+  function onTextAreaKeyDown(e) {
+    var textarea = $(this);
+    var submit = textarea.closest('form').find('.form-submit');
+    if ($.trim(textarea.val()) != '') {
+      if ((e.keyCode === 13 && e.ctrlKey) || (e.keyCode === 13 && e.metaKey)) {
+        e.preventDefault();
+        submit.trigger('click');
```

Comment:
Hey @albertoalaejos,

What is the reason that we click the submit button here instead of just calling `$form.submit()` on the form the textarea belongs to? I think the `.form-submit` class is mostly added for styling in our own theme, which means this Javascript will depend on our theme to function.

If there is a specific reason then we should probably add a comment to document this : )